src/pages/index.js

```diff
@@ -19,64 +21,187 @@ import {
+    });
+});
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
-const editFormValidator = new FormValidator(settings, profileEditForm);
-const addFormValidator = new FormValidator(settings, addCardFormElement);
+function handleLikeClick(data) {
+  if (!data.isLiked) {
+    api
+      .likeACard(data._id, { method: "PUT" })
+      .then((res) => {
+        console.log(res);
+        data.isLiked = true;
+        data.toggleLike();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .dislikeACard(data._id, { method: "DELETE" })
+      .then((res) => {
+        console.log(res);
+        data.isLiked = false;
+        data.toggleLike();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleDeleteClick(card) {
+  const cardId = card._id;
+  if (!cardId) {
+    console.error("card._id is undefined");
+    return;
+  }
+  confirmDeleteModal.setSubmitFunction(() => {
+    api
+      .handleDeleteCard(cardId)
+      .then((res) => {
+        console.log(res);
+        card.handleDeleteCard();
+        confirmDeleteModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  confirmDeleteModal.open();
+}
 
 function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleCardPreview);
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handleCardPreview,
+    handleLikeClick,
+    handleDeleteClick
+  );
   return cardElement.getView();
 }
 
 previewImagePopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileEditSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (userData) => {
+  editProfilePopup.setLoading("Saving...");
+  handleProfileEditSubmit(userData);
+});
 editProfilePopup.setEventListeners();
 
 addCardPopup.setEventListeners();
 
-cardList.renderItems();
+// cardList.renderItems();
 
 function handleCardPreview(cardData) {
   previewImagePopup.open(cardData);
 }
 
 // Event Handlers
 
-function handleProfileEditSubmit({ name, description }) {
-  userInfo.setUserInfo({ name, description });
+function handleProfileEditSubmit(userData) {
+  const { name, description } = userData;
+  api
+    .updateUserProfile({ name, about: description })
+    .then((updateUserData) => {
+      console.log(updateUserData);
+      userInfo.setUserInfo({
+        name: updateUserData.name,
+        description: updateUserData.about,
+      });
+
+      profileEditForm.reset();
+
+      editProfilePopup.close();
+    })
+    .catch((err) => console.log(err));
+}
+
+function handleAvatarEditSubmit(input) {
+  const link = input.avatar;
+  api
+    .editProfileImage(link)
+    .then((userData) => {
+      userInfo.setAvatarInfo(userData.avatar);
+      editAvatarModal.setLoading("Saving...");
```

Comment:
`      editAvatarModal.setLoading("Saving...");
`

You need to start the loading effect **before** the request (not in `then`)